packages/@aws-cdk/aws-lambda-nodejs/lib/bundling.ts

```diff
@@ -112,6 +112,10 @@ export class Bundling implements cdk.BundlingOptions {
       throw new Error('preCompilation can only be used with typescript files');
     }
 
+    if (props.format === OutputFormat.ESM && props.runtime === Runtime.NODEJS_12_X) {
```

Comment:
@jogold I think `NODEJS_10_X` is still supported (at least for updating functions).